lib/msal-browser/src/utils/BrowserUtils.ts

```diff
@@ -20,12 +20,20 @@ export class BrowserUtils {
      * @param {string} urlNavigate - URL of the authorization endpoint
      * @param {boolean} noHistory - boolean flag, uses .replace() instead of .assign() if true
      */
-    static navigateWindow(urlNavigate: string, noHistory?: boolean): void {
+    static navigateWindow(urlNavigate: string, navigationTimeout: number, logger: Logger, noHistory?: boolean): Promise<void> {
         if (noHistory) {
             window.location.replace(urlNavigate);
         } else {
             window.location.assign(urlNavigate);
         }
+
+        // To block code from running after navigation, this should not throw if navigation succeeds
+        return new Promise((resolve) => {
+            setTimeout(() => {
+                logger.warning("Expected to navigate away from the current page but timeout occurred.");
+                resolve();
```

Comment:
Would `resolve(null)` mitigate the need for possibly returning `void` in the public api?

Reply:
(would be fine with that if so)

Reply:
yes, but most places that call `navigateWindow` actually do return void (loginRedirect, logout, RedirectHandler) If we want to not return void from `handleRedirectPromise` we can just await this there then resolve null.

Reply:
> If we want to not return void from handleRedirectPromise we can just await this there then resolve null instead of returning this directly

That works.

Reply:
Done